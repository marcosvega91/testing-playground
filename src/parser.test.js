import parser from './parser';
import { renderIntoDocument } from '../tests/utils/dom-render';

const setup = () => {
  return renderIntoDocument(`
    <div>
      <button>btn1</button>
      <button>btn2</button>
      <span>I'm a label</span>
      <input placeholder="I'm a placeholder" />
    </div>`);
};
describe('parser', () => {
  it('should give an error if the query is not valid', () => {
    const container = setup();
    const result = parser.parse(container, 'invalidquery');
    expect(result.error).toBeDefined();
    expect(result.errorBody).toBeDefined();
  });

  it('should return targets as array', () => {
    const container = setup();
    let result = parser.parse(container, "screen.getAllByRole('button')");
    expect(result.targets).toBeArray();
    expect(result.targets).toHaveLength(2);
    result = parser.parse(container, 'screen.getByText("I\'m a label")');
    expect(result.targets).toBeArray();
    expect(result.targets).toHaveLength(1);
  });

  it('should return getByPlaceholderText as expression', () => {
    const container = setup();
    let result = parser.parse(
      container,
      `screen.getAllByRole('button');
        screen.getByPlaceholderText("I'm a placeholder");`,
    );
    expect(result.expression).toMatchSnapshot();
  });
});
