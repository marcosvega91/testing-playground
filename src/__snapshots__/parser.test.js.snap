// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser should return getByPlaceholderText as expression 1`] = `
Object {
  "args": Array [
    "I'm a placeholder",
  ],
  "call": "getByPlaceholderText(\\"I'm a placeholder\\")",
  "expression": "screen.getByPlaceholderText(\\"I'm a placeholder\\")",
  "level": 0,
  "method": "getByPlaceholderText",
  "scope": "screen",
}
`;
